pkg/cortex/cortex.go

```diff
@@ -247,6 +249,42 @@ func (t *Cortex) init(cfg *Config, m moduleName) error {
 	return t.initModule(cfg, m)
 }
 
+// calls start functions of modules. Functions are called concurrently.
+func (t *Cortex) start(target moduleName) error {
+	deps := orderedDeps(target)
+
+	wg := sync.WaitGroup{}
```

Comment:
Since you are only returning errors you should consider using a [`sync.errgroup`](https://godoc.org/golang.org/x/sync/errgroup#Group.Go) instead of the waitgroup and channel.

Reply:
Thanks, looks exactly like what I need.